src/requestVariableCacheValueProcessor.ts

```diff
@@ -1,114 +1,83 @@
 "use strict";
-import { window } from "vscode";
-
-import { MimeUtility } from "./mimeUtility";
 import { RequestVariableCacheValue } from './models/requestVariableCacheValue';
 import { HttpResponse } from './models/httpResponse';
+import { HttpRequest } from "./models/httpRequest";
+
+const jp = require('jsonpath');
 
-const validPathRegex = new RegExp(/(\w+)(\.\w+|\[\d+\])*/);
-const partRegex = new RegExp(/(\w+)(\[\d+\])*/);
-const arrayIndexRegex = new RegExp(/\[(\d+)\]/);
+const requestVariablePathRegex: RegExp = /(\w+)(?:\.)?(response|request)?(?:\.)?(body|headers)?(?:\.)?(.*)?/;
 
 export class RequestVariableCacheValueProcessor {
     public static getValueAtPath(value: RequestVariableCacheValue, path: string): any {
         if (!value
-            || !path
-            || !RequestVariableCacheValueProcessor.isValidPath(path)) {
+            || !path) {
             return undefined;
         }
 
-        const parts = path.split(".");
+        const matches = path.match(requestVariablePathRegex);
 
-        // Expect first part to be variable name
-        if (parts.length === 1) {
-            return value;
+        if (!matches) {
+            return undefined;
         }
 
-        const [, type, ...rest] = parts;
+        const [, , type, httpPart, rest] = matches;
+
+        if (!type) {
+            return value;
+        }
 
         switch (type) {
             case "request":
-                return RequestVariableCacheValueProcessor.resolveParts(
-                    value.request,
-                    rest
-                );
             case "response":
-                return RequestVariableCacheValueProcessor.resolveResponse(
-                    value.response,
-                    rest
-                );
+                const http = value[type];
+
+                if (!http) {
+                    return undefined;
+                }
+
+                if (!httpPart) {
+                    return http;
+                }
+
+                return RequestVariableCacheValueProcessor.resolveHttpPart(http, httpPart, rest);
             default:
```

Comment:
when will this default branch hit?

Reply:
Probably never; but as you've noted above, regex validation is not infallible. So I'd rather keep.

Reply:
OK